tests/wrappers/test_step_compatibility.py

```diff
@@ -1,35 +1,13 @@
+from functools import partial
+
 import pytest
 
 import gymnasium as gym
-from gymnasium.spaces import Discrete
 from gymnasium.wrappers import StepAPICompatibility
+from tests.generic_test_env import GenericTestEnv, old_step_fn
 
-
-class OldStepEnv(gym.Env):
-    def __init__(self):
-        self.action_space = Discrete(2)
-        self.observation_space = Discrete(2)
-
-    def step(self, action):
-        obs = self.observation_space.sample()
-        rew = 0
-        done = False
-        info = {}
-        return obs, rew, done, info
-
-
-class NewStepEnv(gym.Env):
-    def __init__(self):
-        self.action_space = Discrete(2)
-        self.observation_space = Discrete(2)
-
-    def step(self, action):
-        obs = self.observation_space.sample()
-        rew = 0
-        terminated = False
-        truncated = False
-        info = {}
-        return obs, rew, terminated, truncated, info
+OldStepEnv = partial(GenericTestEnv, step_fn=old_step_fn)
```

Comment:
What benefit does this give us? Instead of a self-contained, easy to understand implementation, now we have to go like "Okay, so it's a partial application, so we call the first argument, let's see what GenericTestEnv is... okay, and the argument is..."

I can definitely agree that it's more clever because of the whole partial application thing, but clear > clever

Reply:
Fair, I will change back